HiP-DataStore/Controllers/QueryHelper.cs

```diff
@@ -51,13 +51,56 @@ public static class QueryHelper
                 : query.Where(x => x.Timestamp > timestamp.Value);
         }
 
-        public static IQueryable<T> FilterByUser<T>(this IQueryable<T> query, ContentStatus status, IIdentity user) where T : ContentBase
+        public static IQueryable<T> FilterByUser<T>(this IQueryable<T> query, ContentStatus status, IIdentity User) where T : ContentBase
         {
-            bool isAllowedGetAll = UserPermissions.IsAllowedToGetAll(user, status);
+            bool isAllowedGetAll = UserPermissions.IsAllowedToGetAll(User, status);
             return query.FilterIf(!isAllowedGetAll, x =>
-                        ((status == ContentStatus.All) && (x.Status == ContentStatus.Published)) || (x.UserId == user.GetUserIdentity()));
+                        ((status == ContentStatus.All) && (x.Status == ContentStatus.Published)) || (x.UserId == User.GetUserIdentity()));
         }
         /// <summary>
+        /// Returns all entries that are located within the radius, defined within the entry itself, around the given latitude and longitude.
+        /// If none or only one coordinate is given, all entries are returned.
+        /// </summary>
+        public static IQueryable<T> FilterByLocation<T>(this IQueryable<T> query, float? latitude, float? longitude) where T : ContentBase
+        {
+            List<int> excludedIds = new List<int>();
+
+            if (typeof(Exhibit) == typeof(T))
+            {
+                List<Exhibit> exhibits = query.ToList().Cast<Exhibit>().ToList();
+
+                if (latitude != null && longitude != null)
+                {
+                    foreach (var entry in exhibits)
+                    {
+
+                        if (GetDistanceFromLatLonInKm(entry.Latitude, entry.Longitude, latitude, longitude) > entry.AccessRadius)
+                        {
+                            excludedIds.Add(entry.Id);
+                        }
+                    }
+                    return FilterByIds(query, excludedIds, null);
+                }
+
+            }
+            return query;
+        }
+
+        /// <summary>
+        /// Calculates the distance between two points of latitude and longitude in km.
+        /// </summary>
+        private static double GetDistanceFromLatLonInKm(float? lat1, float? lon1, float? lat2, float? lon2)
```

Comment:
Are you sure you want the arguments to be of type `float?` (that is, [`Nullable<float>`](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/using-nullable-types))?. In your code you are ultimately casting them to `double`, which might cause an exception if one of the specified arguments is `null`.
Consider using just `float`, which will also make all the `double`-casts unnecessary.

*Pro tip*: Adding `using static System.Math;` to the beginning of the file lets you shorten all `Math.Bla(...)`-calls to just `Bla(...)` 😉